skyportal/services/test_server/test_api_server.py

```diff
@@ -57,6 +57,33 @@ def get_cache_file():
     return current_file
 
 
+def treasuremap_request_matcher(r1, r2):
+    """
+    Helper function to help determine if two requests to the TreasureMap API are equivalent
+    """
+
+    # A request matches a TreasureMap request if the URI and method matches
+
+    r1_uri = r1.uri.replace(":443", "")
+    r2_uri = r2.uri.replace(":443", "")
+
+    def is_treasuremap_request(uri):
+        patterns = {
+            "delete": r"/api/v0/cancel_all/$",
+            "submit": r"/api/v0/pointings/$",
+        }
+        for (submit_type, pattern) in patterns.items():
+            if re.search(pattern, uri) is not None:
+                return submit_type
+
+        return False
+
+    r1_is_treasuremap = is_treasuremap_request(r1_uri)
+    r2_is_treasuremap = is_treasuremap_request(r2_uri)
+
+    assert r1_is_treasuremap == r2_is_treasuremap and r1.method == r2.method
```

Comment:
What happens if both `r1_is_treasuremap` and `r2_is_treasuremap` are `False`?

Reply:
Luckily the "treasuremapness" is also caught in  the individual get, post, etc. calls like:
        elif self.request.uri == "/api/v0/pointings":
            match_on = ["treasuremap"]
So double safety here I guess.